project/publish.sc

```diff
@@ -76,6 +76,34 @@ trait ScalaCliPublishModule extends PublishModule with PublishLocalNoFluff {
         computePublishVersion(state, simple = true)
       }
   }
+  def ghToken() = {
+    Option(System.getenv("UPLOAD_GH_TOKEN"))
+      .getOrElse {
+        sys.error("UPLOAD_GH_TOKEN not set")
+      }
+  }
+  def gitClone(repo: String, branch: String, workDir: os.Path) = {
+    os.proc("git", "clone", repo, "--depth", "1", "-q", "-b", branch).call(cwd = workDir)
+  }
+  def setupGithubRepo(repoDir: os.Path) = {
+    val gitUserName = "Github Actions"
+    val gitEmail    = "actions@github.com"
+
+    os.proc("git", "config", "user.name", gitUserName).call(cwd = repoDir)
+    os.proc("git", "config", "user.email", gitEmail).call(cwd = repoDir)
+  }
+  def commitChanges(name: String, branch: String, repoDir: os.Path): Unit = {
+    if (
+      os.proc("git", "status").call(cwd = repoDir).out.text().trim.contains("nothing to commit")
+    ) {
+      println("Nothing Changes")
+    }
+    else {
+      os.proc("git", "add", "-A").call(cwd = repoDir)
+      os.proc("git", "commit", "-am", name).call(cwd = repoDir)
+      os.proc("git", "push", "origin", branch).call(cwd = repoDir)
+    }
+  }
```

Comment:
Is there a reason to put those in `PublishModule`? I'm not sure it's the right place for those (just leaving them in `build.sc`, or putting them at the root of `settings.sc`, would have worked better IMO).

Reply:
There is no reason to put this logic in `Publish Module`. So I'm going to move this logic to `build.sc` except method `getToken()` which I will move to `settings.sc`.

Reply:
It seems these are still in `ScalaCliPublishModule`, right?

Reply:
Right, I copy the logic to build.sc from settings.sc, but I forgot to remove it from settings.sc